cpp/include/gpu_operations.h

```diff
@@ -48,6 +48,21 @@ namespace Nice {
 template <typename T>
 class GpuOperations {
  public:
+  ///  static Matrix<T> Multiply(const Matrix<T> &a, const T &scalar)
+  ///                   calculates the element by element product of
+  ///                   a matrix and a scalar.
+  ///
+  ///  \param a
+  ///  const Matrix<T> &a : The m x n matrix to be multiplied by the scalar
+  ///  \param b
+  ///  const T &scalar : The scalar to multiply the matirx
+  ///
+  ///  \return
+  ///  static Matrix<T> Multiply(const Matrix<T> &a, const T &scalar)
```

Comment:
I think you could leave out re-claim the function signatures in all of the return statement.